lib/Provider/CustomOAuth2Connect.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace OCA\SocialLogin\Provider;
+
+use Hybridauth\Adapter\OAuth2;
+use Hybridauth\Data;
+use Hybridauth\Exception\UnexpectedApiResponseException;
+use Hybridauth\User;
+
+class CustomOAuth2Connect extends OAuth2
+{
+
+    /**
+     * @return User\Profile
+     * @throws UnexpectedApiResponseException
+     * @throws \Hybridauth\Exception\HttpClientFailureException
+     * @throws \Hybridauth\Exception\HttpRequestFailedException
+     * @throws \Hybridauth\Exception\InvalidAccessTokenException
+     */
+    public function getUserProfile()
+    {
+        $profileFields = $this->trimExplode(',', $this->config->get('endpoints')->get('profile_fields'), true);
```

Comment:
Why not use just
```php
$profileFields = array_filter( 
    array_map('trim', explode(',', $this->config->get('profile_fields')), //I suggest move profile_fields to top level
    function ($val) { return !empty($val) } 
) 
```
instead of creating new method ?

Reply:
Yeah, you're right. I just copied it from another project...